model/labels/labels.go

```diff
@@ -502,11 +596,44 @@ Outer:
 				continue Outer
 			}
 		}
-		res = append(res, l)
+		res.lbls = append(res.lbls, l)
 	}
 	if len(b.add) > 0 { // Base is already in order, so we only need to sort if we add to it.
-		res = append(res, b.add...)
+		res.lbls = append(res.lbls, b.add...)
 		sort.Sort(res)
 	}
 	return res
 }
+
+// SimpleBuilder allows efficient construction of a Labels from scratch.
+type SimpleBuilder struct {
```

Comment:
```suggestion
type ScratchBuilder struct {
```

Perhaps there is a way to name it with more meaningful name?

Reply:
Accepted your suggestion.